python/ray/util/check_serialize.py

```diff
@@ -0,0 +1,205 @@
+            if name.startswith("__") and name.endswith(
+                    "__") or inspect.isbuiltin(obj):
+                continue
+            serializable, _ = inspect_serializability(
+                obj,
+                name=name,
+                parent=parent,
+                depth=depth - 1,
+                _failure_set=failure_set)
+            found = found or not serializable
+            if found:
+                break
+    if not found:
+        _printer.print(
+            f"WARNING: Did not find non-serializable object in {base_obj}. "
+            "This may be an oversight.")
+    return found
+
+
+def inspect_serializability(base_obj,
+                            name=None,
+                            parent=None,
+                            depth=3,
+                            _failure_set=None):
+    """Identifies what objects are preventing serialization.
+
+    Args:
+        base_obj: Object to be serialized.
+        name (str): Optional name of string.
+        depth (int): Depth of the scope stack to walk through. Defaults to 3.
+
+    Returns:
+        bool: True if serializable.
+        set: Set of objects that cloudpickle is unable to serialize.
+
+    .. versionadded:: 1.1.0
+
+    """
+    colorama.init()
+    top_level = False
+    declaration = ""
+    found = False
+    if _failure_set is None:
+        top_level = True
+        _failure_set = set()
+        declaration = f"Checking Serializability of {base_obj}"
+        print("=" * min(len(declaration), 80))
+        print(declaration)
+        print("=" * min(len(declaration), 80))
+
+        if name is None:
+            name = str(base_obj)
+    else:
+        _printer.print(f"Serializing '{name}' {base_obj}...")
+    try:
+        cp.dumps(base_obj)
+        # _printer.print("...pass...")
+        return True, _failure_set
+    except Exception as e:
+        _printer.print(f"{colorama.Fore.RED}!!! FAIL{colorama.Fore.RESET} "
+                       f"serialization: {e}")
+        found = True
+        try:
+            if depth == 0:
+                _failure_set.add(FailureTuple(base_obj, name, parent))
+        except Exception:
+            pass
+
+    if depth <= 0:
+        return False, _failure_set
+
+    # TODO: we only differentiate between 'function' and 'object'
+    # but we should do a better job of diving into something
+    # more specific like a Type, Object, etc.
+    if inspect.isfunction(base_obj):
+        _inspect_func_serialization(
+            base_obj, depth=depth, parent=base_obj, failure_set=_failure_set)
+    else:
+        _inspect_generic_serialization(
+            base_obj, depth=depth, parent=base_obj, failure_set=_failure_set)
+
+    if not _failure_set:
+        _failure_set.add(FailureTuple(base_obj, name, parent))
+
+    if top_level:
+        print("=" * min(len(declaration), 80))
+        if not _failure_set:
+            print("Nothing failed the diagnostic test, though "
```

Comment:
This message sounds a little unclear to me because I don't know what the diagnostic test does.

Reply:
```suggestion
            print("Nothing failed the inspect_serialization test, though "
```